reposerver/repository/repository.go

```diff
@@ -605,6 +606,14 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
 				if err != nil {
 					return nil, err
 				}
+
+				_, err = os.Stat(path)
+				if os.IsNotExist(err) {
+					log.Warnf(" %s values file does not exist", path)
```

Comment:
Most of the warning messages in logs indicate a problem that Argo CD operators have to act on. I think a warning about a missing values file is intended for the end-user only. This warning message will be printed in logs pretty frequently and won't be very useful for operators. 

I would suggest changing the log level to DEBUG. Additionally, we can introduce a "warning" condition instead. (similar to https://github.com/argoproj/argo-cd/blob/cb1f06c72ad4e52b5a8df4d016010d738b3cb1c4/pkg/apis/application/v1alpha1/types.go#L997). In my opinion "warning" condition is a nice to have a feature and I would work on it in a separate PR.

Reply:
Thank you @alexmt for taking the time to start a review and for your comments. 

I agree with the above on changing the log level as this definitely affects the end-user and not the operator. TBH I did think initially to emit an event for the application resource, but your pointing me to the status conditions as an alternative to warn the user makes sense too and I'll explore it (though might leave it for a separate PR depending on how much work it'd represent and other considerations, like the proper warning condition etc.). 
In any case, my first idea was to let the end-user know about missing files, yet when looking at where in the RepoServer the file check takes place and what the various methods invoked return, I opted for the simpler log approach without much thought on how this would affect logs for operators with multiple managed resources. 

To sum up, I'll update my branch, modify the log level *and* the location of the log within the affirmative conditional below (in normal conditions the helm operation will fail and we should have information on the missing file(s) being the cause) and explore ways to update resource status conditions and at least leave a comment here on that.

Reply:
@alexmt after a quick look, it seems to me that the best way to pass back this type of info from the RepoServer would be to add a new field to the [proto ManifestResponse message](https://github.com/argoproj/argo-cd/blob/b77f947e1dde6b200a2044780c466b364818916d/reposerver/repository/repository.proto#L45-L54) for the caller to handle the condition/event update/generation based on that field.
If his sounds reasonable, I'd be willing to take a stab at it. In that case, let me know if you'd like me to create an enhancement first or what would be the best way to proceed.

Reply:
I agree it makes sense to use `ManifestResponse` to pass back additional information. Thank you for resolving the comment about log level! Merging this PR.